sdk/identity/identity/src/credentials/azurePowerShellCredential.ts

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as childProcess from "child_process";
+import { processUtils } from "../util/processUtils";
+import { TokenCredential, GetTokenOptions, AccessToken } from "@azure/core-http";
+import { CredentialUnavailableError } from "../client/errors";
+import { credentialLogger, formatSuccess, formatError } from "../util/logging";
+import { trace } from "../util/tracing";
+import { ensureValidScope, getScopeResource } from "../util/scopeUtils";
+
+const logger = credentialLogger("AzurePowerShellCredential");
+
+/**
+ * Formats a command based on the platform executing it.
+ * @internal
+ */
+function formatCommand(commandName: string): string {
+  if (process.platform === "win32") {
+    return `${commandName}.exe`;
+  } else {
+    return commandName;
+  }
+}
+
+/**
+ * Receives a list of commands to run, executes them, then returns the outputs.
+ * If anything fails, an error is thrown.
+ * @internal
+ */
+function runCommands(commands: string[]): string[] {
+  const result: string[] = [];
+
+  try {
+    commands.forEach((cmd: string) => {
+      result.push(childProcess.execSync(cmd).toString("utf-8"));
```

Comment:
`exec` is really easy to exploit through shell escape tricks with inputs, see https://github.com/Azure/ms-rest-nodeauth/issues/115

`execFile` is safer, if you don't invoke a shell